content/new-routing.md

```diff
@@ -0,0 +1,446 @@
+
+Additionally, the URL could contain an HTTP [**query string**](https://en.wikipedia.org/wiki/Query_string) (the part after an optional `?`). If so, Flow Router will also split it up into named parameters, which it calls `queryParams`.
+
+
+Here are some example URLs and the resulting `params` and `queryParams`:
+
+| URL           | matches pattern? | params          | queryParams
+| ---- | ---- | ---- | ---- |
+| /             | no | | |
+| /about        | no | | |
+| /lists/        | no | | |
+| /lists/eMtGij5AFESbTKfkT | yes | { _id: "eMtGij5AFESbTKfkT"} |  { }
+| /lists/1 | yes | { _id: "1"} | { }
+| /lists/1?todoSort=top | yes | { _id: "1"} | { todoSort: "top" }
+
+
+Note that all of the values in `params` and `queryParams` are always strings since URLs don't have any way of encoding data types. For example, if you wanted a parameter to represent a number, you might need to use `parseInt(value, 10)` to convert it when you access it.
+
+## Accessing Route information
+
+Flow Router makes a variety of information available via (reactive and otherwise) functions on the global singleton `FlowRouter` (this is the same object that we attached routes to above). As the user navigates around your app, the values of these functions will change (reactively in some cases) correspondingly.
+
+Like any other global singleton in your application (see the X article for info about stores), it's best to limit your access to `FlowRouter`. That way the parts of your app with remain modular and more independent. In the case of `FlowRouter`, it's best to access it solely from the top of your component hierarchy, either in the "page" component, or the layouts that wrap it (see below).
+
+### The current route
+
+To access the current route, you can use `FlowRouter.current()`. This is a object representing all aspects of the route, and as it changes often it is not reactive. 
+
+Often it's more useful to access just exactly what parts of the route you care about. Here are some useful reactive functions you can call:
+
+* `FlowRouter.getRouteName()` gets the name of the route
+* `FlowRouter.getParam(paramName)` returns the value of a single URL parameter
+* `FlowRouter.getQueryParam(paramName)` returns the value of a single URL query parameter
+
+So in our example of the list page form the Todos app, we access the current list's id with `FlowRouter.getParam('_id')` (we'll see more on this below).
+
+### Highlighting the active route
+
+One situation where it is sensible to access the global `FlowRouter` singleton to access the current route's information deeper in the component hierarchy is when rendering links via a navigation component. It's often required to highlight the "active" route in some way (this is the route or section of the site that the user is currently looking at).
```

Comment:
You actually tend to only need to know the active route in a few places in your app (typically the nav). So it's not so bad. For the same reason you could be pure about it, although it's a big pain.

Generating links is a huge problem with "purity".

The thing is that a link href is conceptually just an event handler. There's no problem with using global singletons in event handlers. So one solution would be to not actually use the href, and to use a click event instead (but this is crap, because links are good!)

It's actually not *so* bad because the route map is static. So although it's global, it's not stateful. Like it's really just like calling out to a library for rendering (the "router + route map" is the library in this case).

I don't think we are pure enough in Meteor yet to get into these concepts though :) Maybe we'll get there?